src/client/pythonEnvironments/legacyIOC.ts

```diff
@@ -395,8 +408,13 @@ export function registerNewDiscoveryForIOC(
     api: IPythonEnvironments,
     environmentsSecurity: EnvironmentsSecurity,
 ): void {
+    const componentAdapter = new ComponentAdapter(api, environmentsSecurity);
     serviceManager.addSingletonInstance<IComponentAdapter>(
         IComponentAdapter,
-        new ComponentAdapter(api, environmentsSecurity),
+        componentAdapter,
+    );
+    serviceManager.addSingletonInstance<IExtensionSingleActivationService>(
+        IExtensionSingleActivationService,
+        componentAdapter,
     );
```

Comment:
```suggestion
    serviceManager.addBinding(IComponentAdapter, IExtensionSingleActivationService);
```
Should this just be using binding?

Reply:
I'm not sure how binding in container works, have never used it. But it's also used in import tracker for similar purposes so I guess it should work,
```typescript
serviceManager.addBinding(IImportTracker, IExtensionSingleActivationService);
```

Reply:
It does work